frappe/public/js/frappe/assets.js

```diff
@@ -124,19 +124,34 @@ frappe.assets = {
 		// this is virtual page load, only get the the source
 		// *without* the template
 
-		frappe.call({
-			type: "GET",
-			method: "frappe.client.get_js",
-			args: {
-				items: items,
-			},
-			callback: function (r) {
-				$.each(items, function (i, src) {
-					frappe.assets.add(src, r.message[i]);
-				});
-				callback();
-			},
-			freeze: true,
+		if (items.length === 0) {
+			callback();
+			return;
+		}
+
+		const versionString = frappe.boot.developer_mode
+			? Date.now()
+			: frappe.boot.metadata_version;
```

Comment:
We don't need metadata version here right? window._version_number is more suitable for this.